src/overlays/actors/ovl_Bg_Mori_Elevator/z_bg_mori_elevator.c

```diff
@@ -20,37 +34,252 @@ const ActorInit Bg_Mori_Elevator_InitVars = {
     (ActorFunc)BgMoriElevator_Update,
     NULL,
 };
-*/
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern UNK_TYPE D_060035F8;
+extern Gfx D_06002AD0[];
+
+// Couldnt get a mov.s to go into a delay slot
+#ifdef NON_MATCHING
+f32 func_808A1800(f32* posY, f32 target, f32 arg2, f32 yVel, f32 arg4) {
+    f32 phi_f2;
+    phi_f2 = (target - *posY) * arg2;
+    if (*posY < target) {
+        if (yVel < phi_f2) {
+            phi_f2 = yVel;
+        } else {
+            if (phi_f2 < arg4) {
+                phi_f2 = arg4;
+            }
+        }
+        *posY = (*posY + phi_f2);
+
+        if (target < *posY) {
+            *posY = target;
+            return phi_f2;
+        }
+    } else {
+        if (target < *posY) {
+            if (phi_f2 < (-yVel)) {
+                phi_f2 = (-yVel);
+            } else {
+                if ((-arg4) < phi_f2) {
+                    phi_f2 = (-arg4);
+                }
+            }
+            *posY = (*posY + phi_f2);
+            if (*posY < target) {
+                *posY = target;
+                return phi_f2;
+            }
+        } else {
+            phi_f2 = 0.0f;
+            if (phi_f2) {}
+            return phi_f2;
+        }
+    }
+
+    // return phi_f2;
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1800.s")
+#endif
+
+void func_808A18FC(BgMoriElevator* this, f32 distTo) {
+    f32 phi_f12;
+    f32 temp_f2;
+    f32 temp1;
+
+    // clang-format off
+    do {temp_f2 = fabsf(distTo) * 0.09f; if (temp_f2 < 0.0f) { phi_f12 = 0.0f; } else { temp1 = (temp_f2 > 1.0f) ? (1.0f) : (temp_f2);phi_f12 = temp1;} func_800F436C(&this->dyna.actor.projectedPos, 0x2079, phi_f12);} while (0);
```

Comment:
Was this a permuter suggestion? I am almost 100% sure you don't have to put the entire function on one line. I would try to take what the permuter gives you and make it more "normal" until it doesn't match again.